q2_quality_filter/tests/test_filter.py

```diff
@@ -0,0 +1,750 @@
+            min_quality=15,
+            window_length=2,
+            min_length_fraction=0.5,
+            max_ambiguous=0
+        )
+        exp_record = FastqRecord(
+            b'@header', b'ATTCTN', b'+', b'MMLMLL'
+        )
+        exp_status = RecordStatus.TRUNCATED_AMBIGUOUS
+        self.assertEqual(processed_record, exp_record)
+        self.assertEqual(status, exp_status)
+
+        # truncation and too short
+        fastq_record = FastqRecord(
+            b'@header', b'ATTCTGTA', b'+', b'MMLMLL++'
+        )
+        processed_record, status = _process_record(
+            copy(fastq_record),
+            phred_offset=33,
+            min_quality=15,
+            window_length=2,
+            min_length_fraction=0.9,
+            max_ambiguous=0
+        )
+        exp_record = FastqRecord(
+            b'@header', b'ATTCTG', b'+', b'MMLMLL'
+        )
+        exp_status = RecordStatus.SHORT
+        self.assertEqual(processed_record, exp_record)
+        self.assertEqual(status, exp_status)
+
+    def test_is_retained(self):
+        filtering_stats_df = pd.DataFrame(
+            data=0,
+            index=['sample-a', 'sample-b', 'sample-c'],
+            columns=[
+                'total-input-reads',
+                'total-retained-reads',
+                'reads-truncated',
+                'reads-too-short-after-truncation',
+                'reads-exceeding-maximum-ambiguous-bases'
+            ]
+        )
+
+        # retained and truncated
+        retained = _is_retained(
+            forward_status=RecordStatus.TRUNCATED,
+            reverse_status=RecordStatus.UNTRUNCATED,
+            filtering_stats_df=filtering_stats_df,
+            sample_id='sample-a'
+        )
+        self.assertTrue(retained)
+        self.assertEqual(
+            filtering_stats_df.loc['sample-a', 'total-retained-reads'], 1
+        )
+        self.assertEqual(
+            filtering_stats_df.loc['sample-a', 'reads-truncated'], 1
+        )
+        filtering_stats_df.iloc[:, :] = 0
+
+        # forward read only, retained
+        retained = _is_retained(
+            forward_status=RecordStatus.TRUNCATED,
+            reverse_status=None,
+            filtering_stats_df=filtering_stats_df,
+            sample_id='sample-a'
+        )
+        self.assertTrue(retained)
+        self.assertEqual(
+            filtering_stats_df.loc['sample-a', 'total-retained-reads'], 1
+        )
+        self.assertEqual(
+            filtering_stats_df.loc['sample-a', 'reads-truncated'], 1
+        )
+        self.assertEqual(
+            filtering_stats_df.loc[
+                'sample-a', 'reads-too-short-after-truncation'
+            ],
+            0
+        )
+        filtering_stats_df.iloc[:, :] = 0
+
+        # forward read only, short
+        retained = _is_retained(
+            forward_status=RecordStatus.SHORT,
+            reverse_status=None,
+            filtering_stats_df=filtering_stats_df,
+            sample_id='sample-a'
```

Comment:
I suggest changing the sample_id on one of these, just so you know that the info gets written to the write spot. 

```suggestion
            sample_id='sample-b'
```